### ISSUE TITLE
"Features" not respected by "update --all"

### ISSUE THREAD

**Issue by pitrou** [conda/conda#1334]
One of my environments has the "pydbg" feature. However, when running "update --all", it upgrades a pydbg-enabled numpy to a non-pydbg numpy:

```
$ conda update --all
Fetching package metadata: ........
Solving package specifications: .
Package plan for installation in environment /home/antoine/34d:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    llvmlite-0.4.0             |          py34_53         7.4 MB

The following packages will be UPDATED:

    llvmlite:   0.3.0-py34_23  --> 0.4.0-py34_53
    nose:       1.3.4-py34_1   --> 1.3.6-py34_0 
    numpy:      1.9.1-2_dbg    [pydbg] --> 1.9.2-py34_0 
    pip:        6.0.8-py34_0   --> 6.1.1-py34_0 
    python:     3.4.3rc1-1_dbg [pydbg] --> 3.4.3-2_dbg   [pydbg]
    setuptools: 14.0-py34_0    --> 15.2-py34_0  
    sqlite:     3.8.4.1-0      --> 3.8.4.1-1    
```

**Comment by asmeurer** [comment]
How did you install the feature? Is there a `pydbg` metapackage with the track_features installed in the environment?

**Comment by pitrou** [comment]
Yes, of course:

```
$ conda list
# packages in environment at /home/antoine/34d:
#
cudatoolkit               6.0                          p0  
llvmlite                  0.4.0                   py34_53  
markupsafe                0.23                     py34_0  
nose                      1.3.6                    py34_0  
numpy                     1.9.2                     2_dbg  [pydbg]
openssl                   1.0.1k                        1  
pip                       6.1.1                    py34_0  
pydbg                     1.0                           0  
python                    3.4.3                     2_dbg  [pydbg]
readline                  6.2                           2  
setuptools                15.2                     py34_0  
sqlite                    3.8.4.1                       1  
system                    5.8                           2  
tk                        8.5.18                        0  
zlib                      1.2.8                         0  
```

**Comment by asmeurer** [comment]
What binstar channel(s) do these packages come from?

**Comment by pitrou** [comment]
Mine ("antoine") :-) Note that I've since circumvented the issue by uploading an updated numpy build (1.9.2 with pydbg).
https://binstar.org/antoine/

**Comment by asmeurer** [comment]
I can reproduce this and I see why it is happening.

**Comment by mcg1969** [comment]
PR #2000 doesn't fix this, but thanks to @jakirkham's efforts I've identified a fix. Assuming that we all agree that conda should respect feature requests _first_, then maximize versions second, then this should work here.

**Comment by pelson** [comment]
@mcg1969 - I believe you have solved this, right?

**Comment by kalefranz** [comment]
I believe @mcg1969 fixed this with #2036.
